### PR TITLE
Navbar #8

### PR DESCRIPTION
**Pull request by smlarkin**
# added a Nav.js containing a NavBar component
This is wired "to a point" and we must meet over the final implementation ASAP.

### CODE REVIEW

File: client/components/Nav.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import PropTypes from 'prop-types'
+import {connect} from 'react-redux'
+import {Link} from 'react-router-dom'
+import {logout} from '../store'
+
+class NavBar extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.toggle = this.toggle.bind(this)
```
Comment (glebec) [inline client/components/Nav.js]: using `bind` in the constructor is common & idiomatic, but you can also consider using class property syntax with arrow functions instead (see below)

File: client/components/Nav.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    this.state = {
+      isOpen: false
+    }
+  }
+  toggle() {
+    this.setState({
+      isOpen: !this.state.isOpen
+    })
+  }
```
Comment (glebec) [inline client/components/Nav.js]: This function could be written using an arrow, and you wouldn't need to manually write the `bind` in that case.

```js
toggle = () => { /* ... * / }
```

File: client/components/Nav.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+export default connect(mapState, mapDispatch)(NavBar)
+
+/**
+ * PROP TYPES
+ */
+Navbar.propTypes = {
+  handleClick: PropTypes.func.isRequired,
+  isLoggedIn: PropTypes.bool.isRequired
+}
```
Comment (glebec) [inline client/components/Nav.js]: Nice using `propTypes`! 👍

File: client/components/Nav.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+            <Nav className="ml-auto" navbar>
+              <NavItem>
+                <NavLink href="/magic/">Magic</NavLink>
+              </NavItem>
+              <NavItem>
+                <NavLink href="/monsters/">Monsters</NavLink>
+              </NavItem>
+              <NavItem>
+                <NavLink href="/otherworldly/">Otherworldly</NavLink>
+              </NavItem>
```
Comment (glebec) [inline client/components/Nav.js]: I think these three `NavItem`s could be dynamically generated from an array of path-label pairs, which would make this a little more concise. The key would be the path, which obviously must be unique.

File: client/components/Nav.js (hunk 2 above)
Comment (smlarkin) [reply client/components/Nav.js]: Will refactor once we have it integrated properly.  I had just rebuild this nav with a new library and want it wired up before I change the code.  ALSO I was not sure if our boilermaker web-pack supported the last class changes in JS.  Will tweak soon.  Thx.

**Pull request merged**